README.md

```diff
@@ -1 +1,237 @@
-# csp-billing-adapter
\ No newline at end of file
+# CSP Billing Adapter
+
+The billing adapter provides a configurable abstraction layer
+for applications to report billing information to the CSP API.
+It is designed as a plugin oriented Python application using
+[Pluggy](https://pluggy.readthedocs.io/en/stable/) for plugin
+management.
+
+## Plugin Hooks
+
+The core adapter is designed to work with 3 distinct plugin
+layers.
+
+### Storage Layer
+
+This layer handles the storage and backup of adapter billing
+data and status. A plugin implementing the hooks for the storage
+layer is required to implement all hooks in the
+[storage_hookspecs.py](https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/main/csp_billing_adapter/storage_hookspecs.py)
+module.
+
+### CSP Layer
+
+This layer handles metered billing in a CSP and metadata
+generation. The hooks found in
+[csp_hookspecs.py](https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/main/csp_billing_adapter/csp_hookspecs.py)
+should be implemented.
+
+### General Setup Layer
+
+There are three hookspecs to implement for the general setup
+layer. These include **`setup_adapter`**, **`load_defaults`**, and
+**`get_usage_data`**. Details about these hookspecs are found in
+[hookspecs.py](https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/main/csp_billing_adapter/hookspecs.py).
+
+## Configuration
+
+The adapter is configured via a YAML file that is accesible to the
+binary. The config file is used to setup the different intervals for billing
+and usage updates. As well as the billing dimensions which are used to
```

Comment:
Not sure the "As well as" starting the sentence here reads exactly right... However, that may just be me...

```suggestion
and usage updates, as well as the billing dimensions, which are used to
```